src/subscription_manager/managercli.py

```diff
@@ -2671,6 +2672,47 @@ def _list(self, all_overrides, specific_repos):
             print(columnize(names, echo_columnize_callback, *values, indent=2) + "\n")
 
 
+class RoleCommand(CliCommand):
+    def __init__(self):
+        shortdesc = _("Modify system purpose role")
+        super(RoleCommand, self).__init__("role", shortdesc, primary=False)
+        self.parser.add_option("--set", dest="set_role",
+                               help=(_("Set role of system purpose")))
+        self.parser.add_option("--unset", dest="unset_role", action="store_true",
+                               help=(_("Unset role of system purpose")))
+
+    def _validate_options(self):
+        if self.options.set_role and self.options.unset_role:
+            system_exit(os.EX_USAGE, _("Error: Options --set and --unset of role subcommand are mutually exclusive."))
+
+    def _set_role(self, role):
+        ret = save_role_to_syspurpose_metadata(role)
+        if ret:
+            print(_("System Purpose role has been set to: %s") % role)
+        else:
+            print(_("Error: System Purpose role has NOT been set to: %s") % role)
+
+    def _unset_role(self):
+        ret = save_role_to_syspurpose_metadata(None)
+        if ret:
+            print(_("System Purpose role has been unset"))
+        else:
+            print(_("Error: System Purpose role has NOT been unset"))
+
+    def _do_command(self):
+        self._validate_options()
+        if self.options.set_role:
+            self._set_role(self.options.set_role)
+        elif self.options.unset_role:
+            self._unset_role()
+        else:
+            sys_purpose_contents = read_syspurpose()
```

Comment:
I know this method came from the merge, but now when running `subman role`, I get this stacktrace:
```
File "/usr/lib64/python2.7/site-packages/subscription_manager-1.22.1-py2.7-linux-x86_64.egg/subscription_manager/syspurposelib.py", line 116, in read_syspurpose
    syspurpose = SyspurposeStore.read(USER_SYSPURPOSE, raise_on_error=True).contents
TypeError: read() got an unexpected keyword argument 'raise_on_error
```
This `raise_on_error` arg doesn't look like it is defined in the `read()` method at all.